### ISSUE TITLE
Feed Reading with malformed dates

### ISSUE THREAD

**Issue by snomad** [alexdebril/feed-io#289]
Hello,

Reading https://walterfootball.com/rss.xml, that feed produces dates such as 
Tues, 26 May 2020 04:26:00 EST

This causes an exception in FeedIo\Rule\DateTimeBuilder.stringToDateTime()

Specifically, it fails this line

```php
 if (false === strtotime($string)) {  
            throw new \InvalidArgumentException('Impossible to convert date : '.$string);  
 }
```

A few questions

1. Rather than failing completely because one item fails, is their a way to instead collect item errors and turn them into warnings (while returning the rest of the valid items)?

2. Is fixing a date format issue like this feasible? It even fails Carbon

**Comment by alexdebril** [comment]
Hi,
Just before answering the questions, let me bring a bit of context to you. When I started to design feed-io a while ago, I was aware that being compliant to RSS / Atom specifications is not easy, so I never expected that 100% of the feeds will be 100% valid. But I also considered that we should expect a bare minimum from feed publishers : at least a title, a content, a link and a valid published date. 

With time flying by, my notion of "valid date" went from "RFC822 and RFC339 only, people have to comply" to "whatever \DateTime::createFromFormat() can parse". Take the history of this array, it's ridiculous: https://github.com/alexdebril/feed-io/blob/master/src/FeedIo/Rule/DateTimeBuilder.php#L22

So a long time ago, I assume that most of the feeds will comply with the bare minimum but I was wrong. And the closed issues board is filled with errors like the one you've just reported, each time I though while closing it : "why did the developer choose such a format? The most common date formats one are defined in constants, why bothering writing something that doesn't already exist ? Anyway, it's probably the last one ...". And guess what ? At best it's true for 6 months, and then I've got a new one. The last time it was a feed produced in Rust: this thing writes the date with the nanoseconds after the actual time, who cares ? Who on Earth wants this level of precision? The best is: PHP doesn't have any symbol for nanoseconds, I cheated by using "u" and "v" symbols to get two integers instead of one, which is ignored by the code anyway.

And now, another developer writes the day's name on four digits instead of three. Maybe he's afraid that in a close future someone will decide to add new days in the week, I don't know...
Anyway, now it's my running gag to handle new date formats and maybe one day I'll print the array and put it into a frame (I know someone who will disagree to have it in the living room)

So, your questions:

1. You're right, but it's probably a backward-compatibility breaking change, I need to figure this out. 
2. I've found the format but I've forgot it while I was writing this long answer... Don't worry the patch is on its way.

Thanks for your contribution and thanks for reading this, I hope you enjoyed the explanation.
